db/migrate/20180509133508_add_null_constraint_to_book.rb

```diff
@@ -0,0 +1,9 @@
+class AddNullConstraintToBook < ActiveRecord::Migration[5.1]
+  def change
+    change_table(:books) do
+      %i[genre author image title publisher year].each do |attr|
```

Comment:
Always try to keep migrations as simple as possible! In this case use one line for each change 
```
change_column_null :books, :genre, false
change_column_null :books, :author, false
change_column_null :books, :image, false
...
```